django/forms/forms.py

```diff
@@ -206,6 +210,11 @@ def _widget_data_value(self, widget, html_name):
 
     def _html_output(self, normal_row, error_row, row_ender, help_text_html, errors_on_separate_row):
         "Output HTML. Used by as_table(), as_ul(), as_p()."
+        warnings.warn(
+            'django.forms.BaseForm._html_output() is deprecated. '
+            'Please use .render() and .get_context() instead.',
+            RemovedInDjango50Warning,
+        )
```

Comment:
I had a look at this today. Not sure if I was holding it wrong (quite possible) but I couldn't see any difference with different stack levels.

Reply:
There is a difference:
- no `stacklevel`:
  ```
  /django/django/forms/forms.py:213: RemovedInDjango50Warning: django.forms.BaseForm._html_output() is deprecated. Please use .render() and .get_context() instead.
  ```
- `stacklevel=2`:
  ```
  /tickets_projects/ticket_31026/ticket_31026/urls.py:41: RemovedInDjango50Warning: django.forms.BaseForm._html_output() is deprecated. Please use .render() and .get_context() instead.
  ```

It's more complicated with `ErrorList` because `as_*()` methods are usually called on templates, and it's hard to find reasonable `stacklevel`. I would use a class name in the warning, instead of `ErrorList`.